### PR TITLE
bato: init at 0.1.7

### LINKED ISSUE #334953: Package request: bato
**Battery status notifications**

**Metadata**

* homepage URL: https://github.com/doums/bato?tab=readme-ov-file
* source URL: https://github.com/doums/bato?tab=readme-ov-file
* license: Mozilla Public License 2.0
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by HaskellHegemonie**
## Description of changes

Small program to send battery notifications
https://github.com/doums/bato

fixes #334953  
Does not ship with a notification server
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`) **Tested the executable without installing a notification server**
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ba/bato/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  pkgs,
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:2]:
```suggestion
```

File: pkgs/by-name/ba/bato/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "doums";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-i2gw8vXiKutq26ACzkVXH3kED7jAngSv2mNo9P3qXnA=";
+  };
+
+  cargoHash = "sha256-cqrSyZjfBSfF015nwOcRy2rkE4XzFPxJCQZGrqZXLEA=";
+
+  nativeBuildInputs = with pkgs; [
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:21]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/by-name/ba/bato/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  pkgs,
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:6]:
```suggestion
  pkg-config
  cmake
}:
```

File: pkgs/by-name/ba/bato/package.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  cargoHash = "sha256-cqrSyZjfBSfF015nwOcRy2rkE4XzFPxJCQZGrqZXLEA=";
+
+  nativeBuildInputs = with pkgs; [
+    pkg-config
+    cmake
+  ];
+
+  buildInputs = with pkgs; [ libnotify ];
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:26]:
```suggestion
  buildInputs = [ libnotify ];
```

File: pkgs/by-name/ba/bato/package.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  meta = with lib; {
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:28]:
```suggestion
  meta = {
```

File: pkgs/by-name/ba/bato/package.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    description = "Small program to send battery notifications";
+    homepage = "https://github.com/doums/bato";
+    changelog = "https://github.com/doums/bato/releases/tag/v0.1.7";
+    license = with licenses; [ mpl20 ];
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:32]:
```suggestion
    license = with lib.licenses; [ mpl20 ];
```

File: pkgs/by-name/ba/bato/package.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    maintainers = [ ];
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:33]: Not going to add yourself as a maintainer here?

File: pkgs/by-name/ba/bato/package.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    platforms = [ "x86_64-linux" ];
```
Comment (ryand56) [inline pkgs/by-name/ba/bato/package.nix:34]: Why only x86_64?

```suggestion
    platforms = lib.platforms.linux;
```

**Review by ryand56 — COMMENTED** [review]
Welcome to nixpkgs!
A few changes that I recommend you implement.

File: pkgs/by-name/ba/bato/package.nix (hunk 7 above)
Comment (HaskellHegemonie) [reply pkgs/by-name/ba/bato/package.nix:33]: I don't plan on maintaining it
But I can take care of it for 1-2 release cycles, I guess
shouldn't recieve much updates

File: pkgs/by-name/ba/bato/package.nix (hunk 8 above)
Comment (HaskellHegemonie) [reply pkgs/by-name/ba/bato/package.nix]: I Implemented your suggested changes

**Review by ryand56 — APPROVED** [review]

File: pkgs/by-name/ba/bato/package.nix [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    cmake
+  ];
+
+  buildInputs = [ libnotify ];
+
+  meta = {
+    description = "Small program to send battery notifications";
+    homepage = "https://github.com/doums/bato";
+    changelog = "https://github.com/doums/bato/releases/tag/v0.1.7";
+    license = with lib.licenses; [ mpl20 ];
```
Comment (dotlambda) [inline pkgs/by-name/ba/bato/package.nix:34]:
```suggestion
    license = lib.licenses.mpl20;
```

File: pkgs/by-name/ba/bato/package.nix [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  fetchFromGitHub,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "bato";
+  version = "0.1.7";
+
+  src = fetchFromGitHub {
+    owner = "doums";
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/by-name/ba/bato/package.nix:16]:
```suggestion
    repo = "bato";
```
#277994

File: pkgs/by-name/ba/bato/package.nix [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    pkg-config
+    cmake
+  ];
+
+  buildInputs = [ libnotify ];
+
+  meta = {
+    description = "Small program to send battery notifications";
+    homepage = "https://github.com/doums/bato";
+    changelog = "https://github.com/doums/bato/releases/tag/v0.1.7";
```
Comment (dotlambda) [inline pkgs/by-name/ba/bato/package.nix:33]:
```suggestion
    changelog = "https://github.com/doums/bato/releases/tag/v${version}";
```

**Pull request merged**